scipy/signal/ltisys.py

```diff
@@ -6,14 +6,14 @@
 
 __all__ = [  # noqa: F822
     'lti', 'dlti', 'TransferFunction', 'ZerosPolesGain', 'StateSpace',
-    'lsim', 'lsim2', 'impulse', 'impulse2', 'step', 'step2', 'bode',
+    'lsim', 'impulse', 'step', 'bode',
     'freqresp', 'place_poles', 'dlsim', 'dstep', 'dimpulse',
-    'dfreqresp', 'dbode', 's_qr', 'integrate', 'interpolate', 'linalg',
+    'dfreqresp', 'dbode', 's_qr', 'linalg',
     'tf2zpk', 'zpk2tf', 'normalize', 'freqs',
     'freqz', 'freqs_zpk', 'freqz_zpk', 'tf2ss', 'abcd_normalize',
     'ss2tf', 'zpk2ss', 'ss2zpk', 'cont2discrete', 'atleast_1d',
-    'squeeze', 'transpose', 'zeros_like', 'linspace',
-    'nan_to_num', 'LinearTimeInvariant', 'TransferFunctionContinuous',
+    'squeeze', 'transpose', 'linspace',
+    'LinearTimeInvariant', 'TransferFunctionContinuous',
```

Comment:
IIUC we now don't need a few functions in this namespace anymore, which were used to implement `lsim2` etc. However, removing those public imports is a breaking change, and we should deprecate it as usual (and point any users of those functions to where they should import the same functionality from instead).

Reply:
Currently deprecated
```

In [3]: from scipy.signal.ltisys import zeros_like
<ipython-input-3-3cb74b9d5e76>:1: DeprecationWarning: `scipy.signal.ltisys.zeros_like` is deprecated along with the `scipy.signal.ltisys` namespace. `scipy.signal.ltisys.zeros_like` will be removed in SciPy 1.13.0, and the `scipy.signal.ltisys` namespace will be removed in SciPy 2.0.0.
  from scipy.signal.ltisys import zeros_like
```

Reply:
Is that a warning from `main` resp 1.12? On 1.11 I see:
```
>>> from scipy.signal.ltisys import integrate
<stdin>:1: DeprecationWarning: Please use `integrate` from the `scipy.signal` namespace, the `scipy.signal.ltisys` namespace is deprecated.
```
I guess it's fine to remove for 1.13, given that the deprecation was at least there since 1.10 (didn't test further back).

Reply:
It will be in the 1.12 release as part of #18279